### PR TITLE
[CELEBORN-76][FOLLOWUP] fix inFlightCommitRequest counting problem

### PR DESCRIPTION
**Pull request by RexXiong**
# [BUG]/[FEATURE] title

### How was this patch tested?

/cc @related-reviewer

/assign @AngersZhuuuu

### CODE REVIEW

File: client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala [hunk 1]
```diff
@@ -293,57 +294,67 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
...
+                          val workerToRequests = currentBatch.flatMap { request =>
+                            if (request.partition.getPeer != null) {
+                              Seq(request.partition, request.partition.getPeer)
+                            } else {
+                              Seq(request.partition)
+                            }
+                          }.groupBy(_.getWorker)
+                          shuffleCommittedInfo.inFlightCommitRequest.addAndGet(
+                            workerToRequests.size)
+                          return workerToRequests
```
Comment (waitinfuture) [inline client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala:310]: prefer not to use return, just add a else block surrounding Map.empty

**Comment by waitinfuture** [comment]
I don't quite get the point this pr fixes, could you explain more? IMO the inFlightCommitRequest stores the number of commitFiles requests, each request can commit files to multiple workers.

**Comment by waitinfuture** [comment]
> > I don't quite get the point this pr fixes, could you explain more? IMO the inFlightCommitRequest stores the number of commitFiles requests, each request can commit files to multiple workers.
> 
> because it may only be partially success, as stageEnd process waits on shuffleCommittedInfo.inFlightCommitRequest, CommitThreads in batchHandleCommitPartitionExecutors may not even execute, but inFlightCommitRequest may be equal to 0.

Previously we increment by 1  before requestCommitFilesWithRetry and decrements by 1 after returns, I think it's consistent.

**Comment by RexXiong** [comment]
> > > I don't quite get the point this pr fixes, could you explain more? IMO the inFlightCommitRequest stores the number of commitFiles requests, each request can commit files to multiple workers.
> > 
> > 
> > because it may only be partially success, as stageEnd process waits on shuffleCommittedInfo.inFlightCommitRequest, CommitThreads in batchHandleCommitPartitionExecutors may not even execute, but inFlightCommitRequest may be equal to 0.
> 
> Previously we increment by 1 before requestCommitFilesWithRetry and decrements by 1 after returns, I think it's consistent.

the inflight count will be consistent finally. but stage end will lost some commit files in those uncommitted threads

**Review by AngersZhuuuu — APPROVED** [review]
LGTM

**Comment by AngersZhuuuu** [comment]
> > > I don't quite get the point this pr fixes, could you explain more? IMO the inFlightCommitRequest stores the number of commitFiles requests, each request can commit files to multiple workers.
> > 
> > 
> > because it may only be partially success, as stageEnd process waits on shuffleCommittedInfo.inFlightCommitRequest, CommitThreads in batchHandleCommitPartitionExecutors may not even execute, but inFlightCommitRequest may be equal to 0.
> 
> Previously we increment by 1 before requestCommitFilesWithRetry and decrements by 1 after returns, I think it's consistent.

We are calculating the completion of this batch. In the previous code, we can only guarantee that the request of a certain worker in this batch is completed, but it is possible that the request of other workers has not started yet, so it enters the calculation datalost

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**